### PR TITLE
Feature/manual subscriptions

### LINKED ISSUE #655: Allow for manual subscriptions to activities via the admin

### PR DESCRIPTION
**Pull request by mstruijs**
Allows activity owners and admins to subscribe and remove external participants from activities.

Fixes #655

### CODE REVIEW

File: module/Activity/src/Activity/Controller/AdminController.php [hunk 1]
```diff
@@ -88,6 +102,101 @@ public function exportPdfAction()
...
+            $error = $translator->translate('Invalid form');
+            $this->redirectSignupRequest($id, false, $error);
+            return;
+        }
+
+        $formData = $form->getData(\Zend\Form\FormInterface::VALUES_AS_ARRAY);
+        $fullName = $formData['fullName'];
+        unset($formData['fullName']);
+        $email = $formData['email'];
+        unset($formData['email']);
```
Comment (jszanto) [inline module/Activity/src/Activity/Controller/AdminController.php]: Why are these unsets needed?

File: module/Activity/src/Activity/Service/Signup.php [hunk 2]
```diff
@@ -31,15 +33,22 @@ protected function getDefaultResourceId()
         return 'activitySignup';
     }
 
+/*    public function getSignupById(ActivitySignup $signup)
+    {
+        $signUpMapper = $this->getServiceManager()->get('activity_mapper_signup');
+        return $signUpMapper->getSignupById($);
+    }
```
Comment (jszanto) [inline module/Activity/src/Activity/Service/Signup.php]: Is this code still needed?

File: module/Activity/src/Activity/Service/Signup.php [hunk 3]
```diff
@@ -220,13 +296,29 @@ public function signOff(ActivityModel $activity, Member $user)
...
 
+    protected function removeSignUp(ActivitySignup $signup)
+    {
         $em = $this->getServiceManager()->get('Doctrine\ORM\EntityManager');
-        $values = $this->getActivityFieldValueMapper()->getFieldValuesBySignup($signUp);
+        $values = $this->getActivityFieldValueMapper()->getFieldValuesBySignup($signup);
         foreach($values as $value){
             $em->remove($value);
         }
```
Comment (jszanto) [inline module/Activity/src/Activity/Service/Signup.php]: This for loop shouldn't be needed if `cascade` is set in the entity relation in the models.

File: module/Activity/src/Activity/Controller/AdminController.php (hunk 1 above)
Comment (mstruijs) [inline module/Activity/src/Activity/Controller/AdminController.php]: This array, apart from the keys `'fullName'` and `'email'`, contains keys referring to values from additional signup fields. Those need to be passed, but the email and fullName must be passed separately. An alternative is to remove these values in the service, but they have to be removed somewhere.

**Pull request merged**